x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/GetDataFrameJobsStatsAction.java

```diff
@@ -0,0 +1,203 @@
+            out.writeString(id);
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            return null;
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.field(FeatureIndexBuilderJob.ID.getPreferredName(), id);
+            return builder;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(id);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == null) {
+                return false;
+            }
+            if (getClass() != obj.getClass()) {
+                return false;
+            }
+            Request other = (Request) obj;
+            return Objects.equals(id, other.id);
+        }
+    }
+
+    public static class RequestBuilder extends ActionRequestBuilder<Request, Response> {
+
+        protected RequestBuilder(ElasticsearchClient client, GetDataFrameJobsStatsAction action) {
+            super(client, action, new Request());
+        }
+    }
+
+    public static class Response extends BaseTasksResponse implements Writeable, ToXContentObject {
+        private List<DataFrameJobStateAndStats> jobsStateAndStats;
+
+        public Response(List<DataFrameJobStateAndStats> jobsStateAndStats) {
+            super(Collections.emptyList(), Collections.emptyList());
+            this.jobsStateAndStats = jobsStateAndStats;
+        }
+
+        public Response(List<DataFrameJobStateAndStats> jobsStateAndStats, List<TaskOperationFailure> taskFailures,
+                List<? extends FailedNodeException> nodeFailures) {
+            super(taskFailures, nodeFailures);
+            this.jobsStateAndStats = jobsStateAndStats;
+        }
+
+        public Response() {
+            super(Collections.emptyList(), Collections.emptyList());
+        }
+
+        public Response(StreamInput in) throws IOException {
+            super(Collections.emptyList(), Collections.emptyList());
+            readFrom(in);
+        }
+
+        public List<DataFrameJobStateAndStats> getJobsStateAndStats() {
+            return jobsStateAndStats;
+        }
+
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            super.readFrom(in);
+            jobsStateAndStats = in.readList(DataFrameJobStateAndStats::new);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeList(jobsStateAndStats);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field(COUNT.getPreferredName(), jobsStateAndStats.size());
+            // XContentBuilder does not support passing the params object for Iterables
+            builder.field(JOBS.getPreferredName());
```

Comment:
You should be able to simply do `builder.field(JOBS.getPreferredName(), jobsStateAndStats)` here. Unless we need to use `Params` it should be fine.